ground/src/main/java/com/google/android/ground/ui/datacollection/TaskFragment.kt

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.datacollection
+
+import com.google.android.ground.ui.editsubmission.AbstractTaskViewModel
+
+interface TaskFragment {
+  var viewModel: AbstractTaskViewModel
+}
```

Comment:
Can we add generic to the interface so that the implementation can provide a type for the ViewModel instead of having to cast before using it?

Reply:
I went ahead and added the generic to TaskFragment. Now the fragment's don't need to cast the viewModel, but we still currently need to cast the ViewModels in the DataCollectionViewPagerAdapter.

Although I will be fixing this in the near future because I'm going to be making the fragments instantiate their own ViewModels.